project_task_materials_stock/models/project.py

```diff
@@ -0,0 +1,151 @@
+                move.do_unreserve()
+            if move.state in ['waiting', 'confirmed', 'assigned']:
+                move.state = 'draft'
+            move.unlink()
+
+    @api.one
+    def write(self, vals):
+        res = super(Task, self).write(vals)
+        if 'stage_id' in vals:
+            if self.consume_material:
+                self.material_ids.create_stock_move()
+                self.material_ids.create_analytic_line()
+            else:
+                self.unlink_stock_move()
+                self.analytic_line_ids.unlink()
+        return res
+
+    @api.multi
+    def action_assign(self):
+        self.stock_move_ids.action_assign()
+
+    @api.multi
+    def action_done(self):
+        self.stock_move_ids.action_done()
+        pass
+
+
+class ProjectTaskMaterials(models.Model):
+    _inherit = "project.task.materials"
+    stock_move_id = fields.Many2one(
+        comodel_name='stock.move', string='Stock Move')
+    analytic_line_id = fields.Many2one(
+        comodel_name='account.analytic.line', string='Analytic Line')
+
+    def _prepare_stock_move(self):
+        product = self.product_id
+        res = {
+            'product_id': product.id,
+            'name': product.partner_ref,
+            'state': 'confirmed',
+            'product_uom': product.uom_id.id,
+            'product_uos': product.uos_id and product.uos_id.id or False,
+            'product_uom_qty': self.quantity,
+            'product_uos_qty': self.quantity,
+            'origin': self.task_id.name,
+            'location_id':
+            product.product_tmpl_id.property_stock_procurement.id,
+            'location_dest_id': self.env.ref(
+                'stock.stock_location_customers').id,
+        }
+        if product.uos_id and product.uom_id and \
+                (product.uos_id != product.uom_id):
+            precision = self.env['decimal.precision'].precision_get(
+                'Product UoS')
+            res['product_uos_qty'] = float_round(
+                self.quantity * product.uos_coeff, precision_digits=precision)
+        return res
+
+    @api.one
+    def create_stock_move(self):
+        move_id = self.env['stock.move'].create(self._prepare_stock_move())
+        self.stock_move_id = move_id.id
+
+    def _prepare_analytic_line(self):
+        product = self.product_id
+        company_id = self.env['res.company']._company_default_get(
+            'account.analytic.line')
+        journal = self.env.ref(
+            'project_task_materials_stock.analytic_journal_sale_materials')
+        res = {
+            'name': self.task_id.name + ': ' + product.name,
+            'ref': self.task_id.name,
+            'product_id': product.id,
+            'journal_id': journal.id,
+            'unit_amount': self.quantity,
+            'account_id': self.task_id.project_id.analytic_account_id.id,
+            'to_invoice':
+            self.task_id.project_id.analytic_account_id.to_invoice.id,
```

Comment:
This field is provided by _hr_timesheet_invoice_ module, so you shouldn't assume that exists. If you don't want to do a glue module, you can do 2 things:

* See if there's a `_prepare_*` or an onchange method on standard that puts these values via inheritance when an analytic account is passed.
* Use attrgetter method to bypass errors in case there's no such value.